src/components/SettingsDrawer.jsx

```diff
@@ -24,62 +24,73 @@ import {
   Button,
   IconButton
 } from "@mui/material";
+import { styled } from "@mui/system";
 import PropTypes from "prop-types";
 import { useNavigate } from "react-router-dom";
 
 import { useGlobalSetting } from "~/contexts/GlobalSettingProvider";
-import { getSetting, setSetting } from "~/hooks/settingsService";
+import { useTheme } from "~/contexts/ThemeProvider";
+import { getSetting, setSetting } from "~/utils/settingsService";
+
+const StyledButton = styled(Button)({
+  marginTop: "16px"
+});
 
 const SettingsDrawer = ({ open, handleClose }) => {
-  const [themeMode, setThemeMode] = useState(() => getSetting("themeMode", "light"));
-  const [language, setLanguage] = useState(() => getSetting("language", "english"));
-  const [themeColor, setThemeColor] = useState(() => getSetting("themeColor", ""));
-  const [selectedTenant, setSelectedTenant] = useState(() => getSetting("selectedTenant", "1801"));
+  const [settings, setSettings] = useState(() => {
+    return {
+      language: getSetting("language", "english"),
```

Comment:
why we are getting settings with hardcoded values like english,"#808080" ,"1801" ?
Still why do we need these states like if you want to get the theme values it can be taken from theme context as it will get values from localStorage only and you can use the same to update also

Reply:
also you can do it as add a key in global context with name personalSettings and in that save all the needed values.
and in global provide add created a reducer that will update the context state and the localstorage at the same time and on init of GlobalProvide set the value from localStorage if available otherwise use initialState values